packages/react/src/ActionMenu/ActionMenu.tsx

```diff
@@ -100,15 +119,52 @@ const Menu: React.FC<React.PropsWithChildren<ActionMenuProps>> = ({
   })
 
   return (
-    <MenuContext.Provider value={{anchorRef, renderAnchor, anchorId, open: combinedOpenState, onOpen, onClose}}>
+    <MenuContext.Provider
+      value={{
+        anchorRef,
+        renderAnchor,
+        anchorId,
+        open: combinedOpenState,
+        onOpen,
+        onClose,
+        // will be undefined for the outermost level, then false for the top menu, then true inside that
+        isSubmenu: parentMenuContext.isSubmenu !== undefined,
+      }}
+    >
       {contents}
     </MenuContext.Provider>
   )
 }
 
 export type ActionMenuAnchorProps = {children: React.ReactElement; id?: string}
 const Anchor = React.forwardRef<HTMLElement, ActionMenuAnchorProps>(({children, ...anchorProps}, anchorRef) => {
-  return React.cloneElement(children, {...anchorProps, ref: anchorRef})
+  const {onOpen, isSubmenu} = React.useContext(MenuContext)
+
+  const openSubmenuOnRightArrow: React.KeyboardEventHandler<HTMLElement> = useCallback(
+    event => {
+      children.props.onKeyDown?.(event)
+      if (isSubmenu && event.key === 'ArrowRight' && !event.defaultPrevented) onOpen?.('anchor-key-press')
+    },
+    [children, isSubmenu, onOpen],
+  )
+
+  // Add right chevron icon to submenu anchors rendered using `ActionList.Item`
+  const parentActionListContext = useContext(ActionListContainerContext)
+  const thisActionListContext = useMemo(
+    () =>
+      isSubmenu ? {...parentActionListContext, defaultTrailingVisual: <ChevronRightIcon />} : parentActionListContext,
+    [isSubmenu, parentActionListContext],
+  )
```

Comment:
I wonder if we need the full context here? 

Could we do something like?

```
<ActionListContainerContext.Provider value={isSubmenu ? {defaultTrailingVisual: <ChevronRightIcon />} : {}}>
      {React.cloneElement(children, {
        ...anchorProps,
        ref: anchorRef,
        onKeyDown: openSubmenuOnRightArrow,
      })}
    </ActionListContainerContext.Provider>
```

We should still memoise it and not inline but just as a pseudo code. Let me know if I underestimate any of the existing context though or missunderstand anything!

Reply:
Yeah, I think we do need the full context because this anchor is still a list item inside a parent list, and so it still needs the info from that list. We just want to override the trailing visual, not the entire context, and React won't merge them by default